File: CS/WindowsFormsApplication1/Form1.cs
using DevExpress.XtraPrinting;
using DevExpress.XtraReports.UI;
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.IO;
using System.Linq;
using System.Text;
using System.Windows.Forms;

namespace WindowsFormsApplication1
{
    public partial class Form1 : Form
    {
        public string FilePostfix { get; set; }
        public Form1()
        {
            InitializeComponent();

            FilePostfix = "Test";
        }

        private void button1_Click(object sender, EventArgs e)
        {
            XtraReport1 report = new XtraReport1();
            ReportPrintTool pt = new ReportPrintTool(report);

            pt.PrintingSystem.AddCommandHandler(new ExcelExportCommandHandler(FilePostfix));
            pt.ShowPreviewDialog();
        }
    }

    public class ExcelExportCommandHandler : ICommandHandler
    {
        public string FileName { get; set; }
        public ExcelExportCommandHandler()
        {

        }
        public ExcelExportCommandHandler(string fileName)
        {
            FileName = fileName;
        }
        public bool CanHandleCommand(PrintingSystemCommand command, IPrintControl printControl)
        {
            return command == PrintingSystemCommand.ExportXlsx;
        }

        public void HandleCommand(PrintingSystemCommand command, object[] args, IPrintControl printControl, ref bool handled)
        {
            if (!CanHandleCommand(command, printControl))
                return;

            XlsxExportOptions options = new XlsxExportOptions() { ExportMode = XlsxExportMode.DifferentFiles} ;

            DialogResult dr = ExportOptionsTool.EditExportOptions(options, printControl.PrintingSystem);
            if (dr == System.Windows.Forms.DialogResult.OK)
            {
                SaveFileDialog sfd = new SaveFileDialog();
                sfd.FileName = "XtraReport.xlsx";

                sfd.Filter = "XLSX File|*.xlsx";
                if (sfd.ShowDialog() == System.Windows.Forms.DialogResult.OK)
                {
                    printControl.PrintingSystem.ExportToXlsx(sfd.FileName, options);

                    string fileName = Path.GetFileNameWithoutExtension(sfd.FileName);
                    string path = Path.GetDirectoryName(sfd.FileName);

                    List<string> fileNames = new List<string>();
                    for (int i = 1; i <= printControl.PrintingSystem.PageCount; i++)
                    {
                        string genFileName = string.Format("{0}\\{1}{2}", path, fileName, i);
                        System.IO.File.Move(genFileName + ".xlsx", String.Format("{0}{1}.xlsx", genFileName, FileName));
                    }                    
                }
            }

            handled = true;
        }
    }
}

File: VB/WindowsFormsApplication1/Form1.vb
Imports DevExpress.XtraPrinting
Imports DevExpress.XtraReports.UI
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.IO
Imports System.Linq
Imports System.Text
Imports System.Windows.Forms

Namespace WindowsFormsApplication1
    Partial Public Class Form1
        Inherits Form

        Public Property FilePostfix() As String
        Public Sub New()
            InitializeComponent()

            FilePostfix = "Test"
        End Sub

        Private Sub button1_Click(ByVal sender As Object, ByVal e As EventArgs) Handles button1.Click
            Dim report As New XtraReport1()
            Dim pt As New ReportPrintTool(report)

            pt.PrintingSystem.AddCommandHandler(New ExcelExportCommandHandler(FilePostfix))
            pt.ShowPreviewDialog()
        End Sub
    End Class

    Public Class ExcelExportCommandHandler
        Implements ICommandHandler

        Public Property FileName() As String
        Public Sub New()

        End Sub
        Public Sub New(ByVal fileName As String)
            Me.FileName = fileName
        End Sub
        Public Function CanHandleCommand(ByVal command As PrintingSystemCommand, ByVal printControl As IPrintControl) As Boolean Implements ICommandHandler.CanHandleCommand
            Return command = PrintingSystemCommand.ExportXlsx
        End Function

        Public Sub HandleCommand(ByVal command As PrintingSystemCommand, ByVal args() As Object, ByVal printControl As IPrintControl, ByRef handled As Boolean) Implements ICommandHandler.HandleCommand
            If Not CanHandleCommand(command, printControl) Then
                Return
            End If

            Dim options As New XlsxExportOptions() With {.ExportMode = XlsxExportMode.DifferentFiles}

            Dim dr As DialogResult = ExportOptionsTool.EditExportOptions(options, printControl.PrintingSystem)
            If dr = System.Windows.Forms.DialogResult.OK Then
                Dim sfd As New SaveFileDialog()
                sfd.FileName = "XtraReport.xlsx"

                sfd.Filter = "XLSX File|*.xlsx"
                If sfd.ShowDialog() = System.Windows.Forms.DialogResult.OK Then
                    printControl.PrintingSystem.ExportToXlsx(sfd.FileName, options)


                    Dim fileName_Renamed As String = System.IO.Path.GetFileNameWithoutExtension(sfd.FileName)
                    Dim path As String = System.IO.Path.GetDirectoryName(sfd.FileName)

                    Dim fileNames As New List(Of String)()
                    For i As Integer = 1 To printControl.PrintingSystem.PageCount
                        Dim genFileName As String = String.Format("{0}\{1}{2}", path, fileName_Renamed, i)
                        System.IO.File.Move(genFileName & ".xlsx", String.Format("{0}{1}.xlsx", genFileName, FileName))
                    Next i
                End If
            End If

            handled = True
        End Sub
    End Class
End Namespace
